velox/core/PlanNode.cpp

```diff
@@ -443,6 +443,109 @@ PlanNodePtr AggregationNode::create(const folly::dynamic& obj, void* context) {
       deserializeSingleSource(obj, context));
 }
 
+namespace {
+RowTypePtr getExpandOutputType(
+    const std::vector<std::vector<TypedExprPtr>>& projections,
+    std::vector<std::string>& names) {
+  VELOX_USER_CHECK(!names.empty());
+  VELOX_USER_CHECK(!projections.empty());
+  for (int32_t i = 0; i < projections.size(); i++) {
+    VELOX_USER_CHECK_EQ(names.size(), projections[i].size());
+  }
+
+  std::vector<TypePtr> types;
+  types.reserve(names.size());
+  for (int32_t i = 0; i < names.size(); ++i) {
+    types.push_back(projections[0][i]->type());
+  }
+
+  return ROW(std::move(names), std::move(types));
+}
+} // namespace
+
+ExpandNode::ExpandNode(
+    PlanNodeId id,
+    std::vector<std::vector<TypedExprPtr>> projections,
+    std::vector<std::string> names,
+    PlanNodePtr source)
+    : PlanNode(std::move(id)),
+      sources_{source},
+      outputType_(getExpandOutputType(projections, names)),
+      projections_(std::move(projections)) {
+  const auto& projectionNames = outputType_->names();
+  const auto numColumns = projectionNames.size();
+  const auto numRows = projections_.size();
+
+  VELOX_USER_CHECK_GT(numColumns, 0);
+  VELOX_USER_CHECK_GT(numRows, 0);
```

Comment:
This check happens to late. getExpandOutputType would already have failed if numRows is zero. Move both these checks to getExpandOutputType.